src/component/ListItem.js

```diff
@@ -0,0 +1,175 @@
+import React, { Component } from 'react';
+
+import '../App.css';
+
+class ListItem extends Component {
+
+  constructor() {
+    super();
+    this.state = {
+      edit: false,
+      editedTodoContent: ''
+    };
+  }
+
+  updateStateContent = e => {
+    const editedTodoContent = e.target.value;
+    this.setState({
+      editedTodoContent
+    });
+  };
+
+  handleSave = () => {
+    const todoItemId = this.props.todoItem.id;
+    if (this.state.editedTodoContent.trim() !== '') {
+      this.props.editTodoItem(todoItemId, this.state.editedTodoContent);
+      this.setEdit(false);
+    } else {
+      alert('Please change your todo content. \nOr press cancel button.');
+    }
+  };
+
+  deleteItem = e => {
+    e.stopPropagation();
+    this.props.deleteTodoItem(this.props.todoItem.id);
+  };
+
+  setEdit = bool => {
+    this.setState({
+      edit: bool
+    });
+  };
+
+  handelEscEnterPress = e => {
```

Comment:
it's `handle...` and not `handel...` :smile:

Reply:
:D